### ISSUE TITLE
`missing_inline_in_public_items` warns about `#[derive]`-generated code

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust-clippy#4861]
Example:

```rust
#![warn(clippy::missing_inline_in_public_items)]

#[derive(Debug)]
pub struct S {}
```

```
warning: missing `#[inline]` for a method
 --> src/lib.rs:3:10
  |
3 | #[derive(Debug)]
  |          ^^^^^
```

I don't think Clippy should warn about this, since the user cannot control the code in question.

**Comment by llogiq** [comment]
This lint just needs a macro check.
